src/dist.rs

```diff
@@ -315,13 +335,58 @@ impl Dist {
     }
 }
 
+#[cfg(feature = "sass")]
+fn sass(
+    static_dir: &std::path::Path,
+    dist_dir: &std::path::Path,
+    options: sass_rs::Options
+) -> Result<()> {
+    fn is_sass(path: &std::path::Path) -> bool {
+        matches!(
+            path.extension().map(|x| x.to_str()).flatten(),
+            Some("sass") | Some("scss")
+        )
+    }
+
+    fn should_ignore(path: &std::path::Path) -> bool {
+        path
+            .file_name()
+            .expect("static file ends by `..`")
+            .to_str()
+            .map(|x| x.starts_with("_"))
+            .unwrap_or(false)
+    }
+
+    log::trace!("Generating dist artifacts");
+    let walker = walkdir::WalkDir::new(&static_dir).into_iter();
+    for entry in walker {
+        let entry = entry.context(format!("cannot walk into directory `{}`", &static_dir.display()))?;
+        let source = entry.path();
+        let dest = dist_dir.join(source.strip_prefix(&static_dir).unwrap());
+        let _ = fs::create_dir_all(dest.parent().unwrap());
+
+        if is_sass(source) && !should_ignore(source) {
+            let dest = dest
+                .with_extension("css");
+
+            let css = sass_rs::compile_file(source, options.clone())
+                .expect("could not convert SASS/ file");
+            fs::write(&dest, css).context(format!("could not write CSS to file `{}`", dest.display()))?;
+        } else if source.is_file(){
+            fs::copy(source, &dest).context(format!("cannot move `{}` to `{}`", source.display(), dest.display()))?;
+        }
```

Comment:
1. You should probably skip directories with extension `.sass`
2. Ignored sass file should not be copied xD
3. use `with_context()` to avoid unnecessary allocations

```suggestion
        if !source.is_file() {
            continue
        } if is_sass(source) {
            if !should_ignore(source) {
                let dest = dest
                    .with_extension("css");
    
                let css = sass_rs::compile_file(source, options.clone())
                    .expect("could not convert SASS/ file");
                fs::write(&dest, css).with_context(|| format!("could not write CSS to file `{}`", dest.display()))?;
            }
        } else {
            fs::copy(source, &dest).with_context(|| format!("cannot move `{}` to `{}`", source.display(), dest.display()))?;
        }
```